### PR TITLE
Skip annotate boolean input

### PR DESCRIPTION
**Pull request by cccclai**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #2957

Differential Revision: [D55946526](https://our.internmc.facebook.com/intern/diff/D55946526/)

### CODE REVIEW

**Comment by chiwwang** [comment]
@shewu-quic @haowhsu-quic

**Comment by shewu-quic** [comment]
Look good for me. Thanks for your help.

File: backends/qualcomm/quantizer/utils.py [hunk 1]
```diff
@@ -41,6 +42,14 @@ def decorator(annotator: Callable):
 
     return decorator
 
+def _is_input_float_tensor(node: Node):
+    """Check if the input is not a float tensor, so that we can skip quantization for the node
+    since observers only works with float Tensors
+    """
+    if "val" not in node.meta or not isinstance(node.meta["val"], FakeTensor):
+        return True
```
Comment (jerryzh168) [inline backends/qualcomm/quantizer/utils.py:50]: why is this True?

File: backends/qualcomm/quantizer/utils.py [hunk 2]
```diff
@@ -123,11 +132,11 @@ def annotate_binary(node: Node, quantization_config: QuantizationConfig) -> None
 
     input_qspec_map = {}
     input_act0 = node.args[0]
-    if isinstance(input_act0, Node):
+    if isinstance(input_act0, Node) and _is_input_float_tensor(input_act0):
```
Comment (jerryzh168) [inline backends/qualcomm/quantizer/utils.py:135]: nit: we could also move `isinstance(input_act0, Node)` check into `_is_input_float_tensor` function as well

**Review by jerryzh168 — APPROVED** [review]

**Pull request closed without merging**